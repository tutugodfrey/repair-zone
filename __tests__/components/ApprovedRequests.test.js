import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ConnectedApprovedRequest, { ApprovedRequests, mapStateToProps }
  from '../../client/src/components/ApprovedRequests.jsx';

let wrapper;
let mockStore = configureStore();
let store = mockStore();

describe('<ApprovedRequest />', () => {
  describe('unconnected component test', () => {
    let wrapper;
    let requests = [
      {
        user: {
          fullname: 'fullname',
          isAdmin: false,
        },
        request: {
          id: 1,
          status: 'pending',
        },
      }
    ];
    const dispatch = jest.fn();
    beforeEach(() => {
      wrapper = shallow(
        <ApprovedRequests
          requests={requests}
          reducedRequests={[]}
          userData={{}}
          dispatch={dispatch}
        />
      );
    });
    test('should map the snapshot for the component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('connected component test', () => {
    let wrapper;
    let requests = [
      {
        user: {
          fullname: 'fullname',
          isAdmin: false,
        },
        request: {
          id: 1,
          status: 'pending',
        },
      }
    ];
    const userData = {
      id: 2,
      fullname: {},
      token: 'tokenvalue',
    }
    store.getState().requests = requests;
    store.getState().userData = userData;
    store.getState().reducedRequests = [];
    beforeEach(() => {
      wrapper = mount(
        <Provider store={store}>
          <ConnectedApprovedRequest />
        </Provider>
      );
    });
    test('should not have a button', () => {
      const btns = wrapper.find('button');
      expect(btns.length).toBe(4)
    })
  });
  describe('mapStateToProps', () => {
    test('should map the state of the component correctly', () => {
      const appState = {
        userData: {},
        reducedRequests: [],
        requests: [],
        errorStatus: true,
        errorMessage: 'ssssss'
      };
      const expectedState = {
        userData: {},
        reducedRequests: [],
        requests: [], 
      };
      const componentState = mapStateToProps(appState);
      expect(componentState).toEqual(expectedState);
    });
  });
});
