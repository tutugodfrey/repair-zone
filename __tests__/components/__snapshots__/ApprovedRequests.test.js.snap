// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApprovedRequest /> unconnected component test should map the snapshot for the component 1`] = `
<div>
  <Connect(Modal) />
  <div>
    <h1
      className="text-center pt-5"
    >
      Approved Requests
    </h1>
    <Div
      content={
        Array [
          <div
            className="view-request-div col-10 offset-1 p-0 mb-1 bg-white col-md-6 offset-md-3"
          >
            <div
              className="request-header d-flex flex-row justify-content-between rounded border px-2 py-2 bg-white"
            >
              <strong
                className="font-weight-bold"
              >
                fullname
              </strong>
              <Button
                buttonClass="view-btn rounded text-white bg-success"
                buttonName="view"
                onClick={[Function]}
              />
            </div>
            <div
              className="request-content d-none p-3"
            >
              <dl
                className="request-detail-dl"
              >
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="issueData-term"
                  >
                    ISSUEDATA
                  </dt>
                  <dd
                    className="issueData-def"
                  >
                    2018-06-17T23:00:00.000Z
                  </dd>
                </div>
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="category-term"
                  >
                    CATEGORY
                  </dt>
                  <dd
                    className="category-def"
                  >
                    electrical
                  </dd>
                </div>
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="description-term"
                  >
                    DESCRIPTION
                  </dt>
                  <dd
                    className="description-def"
                  >
                    my wall socket got burnt and need replacement
                  </dd>
                </div>
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="address-term"
                  >
                    ADDRESS
                  </dt>
                  <dd
                    className="address-def"
                  >
                    provide for user making request
                  </dd>
                </div>
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="urgent-term"
                  >
                    URGENT
                  </dt>
                  <dd
                    className="urgent-def"
                  >
                    true
                  </dd>
                </div>
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="status-term"
                  >
                    STATUS
                  </dt>
                  <dd
                    className="status-def"
                  >
                    pending
                  </dd>
                </div>
              </dl>
              <Button
                buttonClass="reject-request-button px-3 mr-2  rounded bg-danger text-white"
                buttonId="request-1"
                buttonName="Reject"
                onClick={[Function]}
              />
              <Button
                buttonClass="resolve-request-button px-3 mr-2 rounded bg-success text-white"
                buttonId="request-1"
                buttonName="Resolve"
                onClick={[Function]}
              />
            </div>
          </div>,
        ]
      }
      divClass="row py-5"
    />
  </div>
</div>
`;
